source/common/router/config_impl.cc

```diff
@@ -65,6 +67,82 @@ Optional<uint64_t> HashPolicyImpl::generateHash(const Http::HeaderMap& headers)
   return hash;
 }
 
+HeaderFormatterSharedPtr RequestHeaderParser::parseInternal(const std::string& format) {
+	std::string variable_name;
+	if (format.find("%") == 0) {
```

Comment:
@htuch the AccessLogFormatParser::parse supports a lot of formatting like parsing the request and response headers, supports multiple variables in a single format string.
For the custom request header, we do not have request header because the upstream cluster already gets all of that.
So it is a very small piece of parsing code that seems to be common between these parsers and hence kept it separate.